string predicate
string[] arg
---
prolog_interface/solution[] ris